working/0546-patterns/records-feature-specification.md

```diff
@@ -283,6 +283,41 @@ fields are) and collection literals.
 
 **TODO: Specify this more precisely.**
 
+### Constants
+
+A record expression is a compile-time constant expression
+if all its record field expressions are compile-time constant expressions. 
+
+_This is true whether the expression occurs in a constant context or not,
+which means that a record expression can be used directly as a parameter default value 
+if its record field expressions are constant expressions.
+Example: `f({(int, int) x = (1, 2)}) => ...`._
+
+A record expression is a potentially constant expression 
+if all its record field expressions are potentially constant or constant expressions.
+
+_This means that a record expression can be used in the initializer list
+of a constant non-redirecting generative constructor._
+
+Constant *object* instantiations create deeply immutable and canonicalied objects.
```

Comment:
Looking at the rest of the document, it doesn't seem to be as strongly separating rationale from normative text. 
I tried making everything except paragraph 1 and 3 italic, and it looked weird.

Reply:
I think that's OK. It can look weird in this context, but that's just because we ought to change the rest of this document similarly, and in the meantime there's nothing lost by doing it here.

Reply:
I try to use italics for non-normative text when it's in the middle of otherwise clearly normative sections, mostly as a signal to implementers that they don't have to treat it as gospel. But in introductory sections that are higher level or motivational, almost the whole thing is non-normative, so I don't bother highlighting it.

I admit I could probably be clearer and more disciplined about this. :)